### PR TITLE
Rename parse-quil and parse-quil-string

### LINKED ISSUE #146: PARSE-QUIL vs PARSE-QUIL-STRING
These should be named a little bit better.

### PR DESCRIPTION
**Pull request by notmgsk**
parse-quil -> parse-quil-into-raw-program
parse-quil-string -> parse-quil

Closes #146

### CODE REVIEW

File: src/cl-quil.lisp [hunk 1]
```diff
@@ -6,22 +6,31 @@
...
 (defvar *standard-post-process-transforms*
   '(process-includes resolve-applications expand-circuits type-check)
-  "The standard transforms that are applied by PARSE-QUIL-STRING.")
+  "The standard transforms that are applied by PARSE-QUIL.")
 
-(defun parse-quil-string (string &optional originating-file)
-  "Parse and process the Quil string STRING, which originated from the file ORIGINATING-FILE."
-  (let ((pp (parse-quil string)))
-    (dolist (xform *standard-post-process-transforms* pp)
+(defun parse-quil2 (string &key originating-file (transforms *standard-post-process-transforms*))
```
Comment (ecpeterson) [inline src/cl-quil.lisp]: Do we really need both of these variants? :(

Comment (notmgsk) [reply src/cl-quil.lisp]: No that's just me being a dumb dumb

Comment (notmgsk) [reply src/cl-quil.lisp]: A tired dumb dumb

**Comment by stylewarning** [comment]
Could we also have a companion QVM PR?

**Pull request merged**